prov/util/src/util_mr_map.c

```diff
@@ -301,3 +303,23 @@ int ofi_mr_verify(struct ofi_mr_map *map, ssize_t len,
 	fastlock_release(&domain->lock);
 	return ret;
 }
+
+enum fi_hmem_iface ofi_get_mr_device_iface(struct ofi_mr_map *map, void **desc,
+					   uint64_t *device)
+{
+	struct util_domain *domain;
+	struct ofi_mr *mr;
+
+	domain = container_of(map, struct util_domain, mr_map);
+	if (!(domain->mr_mode & FI_MR_HMEM) || !desc || !*desc)
+		return FI_HMEM_SYSTEM;
+
+	mr = (struct ofi_mr *) ofi_mr_map_get(map, (uintptr_t) *desc);
```

Comment:
I don't understand the commit comment associated with these changes.  This lookup deals with handling registrations for RMA target access, not MR caching.  MR caching isn't being used by this patch at all AFAICT.

Reply:
I'm not sure I follow your question. This is basically a way to lookup the MR registration if a provider is using the utility MR registration code. The regattr calls insert the ofi_mr into the domain->mr_map and this function is allowing providers to get the interface and device that the memory was registered with. But as I'm typing this out I realize now what you're saying and that on the initiating side with the desc, we should just be able to cast to the ofi_mr *, is that right? Is that what the desc is for the utility registration?

Reply:
I'm saying that github doesn't let me comment directly on the commit message itself.  That talks about the MR cache.  These changes aren't related to the MR cache.

But, yes, on the initiator side, we can cast the desc to whatever the structure is that we're pointing it at.

Reply:
Ah yes. I see. Will fix all of the things.